Sources/Workspace/Workspace.swift

```diff
@@ -1029,51 +1030,52 @@ extension Workspace {
             }
 
             // Root packages are always allowed to use unsafe flags.
-            result.formUnion(root.packageRefs)
+            result.formUnion(self.root.packageRefs)
 
             return result
         }
 
         func computePackageURLs() -> (required: Set<PackageReference>, missing: Set<PackageReference>) {
-            let manifestsMap: [String: Manifest] = Dictionary(uniqueKeysWithValues:
-                root.manifests.map({ (PackageReference.computeIdentity(packageURL: $0.url), $0) }) +
-                dependencies.map({ (PackageReference.computeIdentity(packageURL: $0.manifest.url), $0.manifest) }))
+            let manifestsMap: [PackageIdentity: Manifest] = Dictionary(uniqueKeysWithValues:
+                root.manifests.map { (PackageIdentity($0.url), $0) } +
+                    self.dependencies.map { (PackageIdentity($0.manifest.url), $0.manifest) }
+            )
 
             var inputIdentities: Set<PackageReference> = []
-            let inputNodes: [GraphLoadingNode] = root.manifests.map({ manifest in
-                let identity = PackageReference.computeIdentity(packageURL: manifest.url)
-                let package = PackageReference(identity: identity, path: manifest.url, kind: manifest.packageKind)
+            let inputNodes: [GraphLoadingNode] = self.root.manifests.map { manifest in
+                let identity = PackageIdentity(manifest.url)
+                let package = PackageReference(identity: identity.description, path: manifest.url, kind: manifest.packageKind)
```

Comment:
can we use the actual identity instead of its description?

Reply:
See https://github.com/apple/swift-package-manager/pull/3032#discussion_r520751255

Reply:
https://github.com/apple/swift-package-manager/pull/3032#discussion_r521326243